### PR TITLE
Fix Knex search

### LINKED ISSUE #1939: Search is not working
# Bug report

## Describe the bug
Keystone list level 'Search' is not working even after adding filters

## Expected behaviour
With filters and without filters List level Search should work checking the entire items in the list.

## System information

- OS: [ macOS, Windows]
- Browser [chrome, safari]

### PR DESCRIPTION
**Pull request by MadeByMike**
This resolves #1939 and #1654 by making the search work the same as the Mongoose adapter.

It's not a good search and more work is needed. See: #343 and #640

### CODE REVIEW

File: packages/adapter-knex/lib/adapter-knex.js [hunk 1]
```diff
@@ -507,6 +511,13 @@ class QueryBuilder {
 
     this._addWheres(w => this._query.andWhere(w), listAdapter, where, baseTableAlias);
 
+    // TODO: Implement configurable search fields for lists
+    const searchField = listAdapter.fieldAdaptersByPath['name'];
+    if (search !== undefined && searchField && searchField.fieldName === 'Text') {
+      this._query.whereRaw('true');
```
Comment (timleslie) [inline packages/adapter-knex/lib/adapter-knex.js]: We don't need the `whereRaw('true')` here, as we're now in the phase of the query builder where we're chaining together `andWhere` calls.

File: packages/adapter-knex/lib/adapter-knex.js [hunk 2]
```diff
@@ -507,6 +511,13 @@ class QueryBuilder {
...
+      this._query.andWhere(`${baseTableAlias}.name`, `LIKE`, `%${search}%`);
```
Comment (timleslie) [inline packages/adapter-knex/lib/adapter-knex.js]: We need to apply `escapeRegExp` to `search` to ensure special characters in the search don't get interpreted as regex controls.

File: packages/adapter-knex/lib/adapter-knex.js [hunk 3]
```diff
@@ -507,6 +511,13 @@ class QueryBuilder {
 
     this._addWheres(w => this._query.andWhere(w), listAdapter, where, baseTableAlias);
 
+    // TODO: Implement configurable search fields for lists
+    const searchField = listAdapter.fieldAdaptersByPath['name'];
+    if (search !== undefined && searchField && searchField.fieldName === 'Text') {
```
Comment (timleslie) [inline packages/adapter-knex/lib/adapter-knex.js]: I'm not sure that `searchField.fieldName === 'Text'` will actually work? TBH I'm not sure, but we should really be checking `searchField.field instanceof Text` to make this more explicit.

Comment (MadeByMike) [reply packages/adapter-knex/lib/adapter-knex.js]: I tested it with different field types but happy to go with instanceof

**Pull request merged**